internal/server/singleprocess/poll.go

```diff
@@ -33,6 +33,12 @@ type pollHandler interface {
 	// is used to define the job returned.
 	PollJob(hclog.Logger, interface{}) (*pb.QueueJobRequest, error)
 
+	// GeneratePollJobs will build a list of QueueJobRequests through PollJob to
+	// schedule for execution. If a pollItem has multiple elements that need to
+	// be polled for, this is required for scheduling a job for each poll item.
+	// If there is only one poll item, the list should only contain one queue job request.
+	GeneratePollJobs(hclog.Logger, interface{}) ([]*pb.QueueJobRequest, error)
```

Comment:
Should we just have `PollJob` return a slice? I don't thikn we need two methods for this: existing things can return just one queue job request.